### PR TITLE
Working produce

### PR DESCRIPTION
**Pull request by andrewhojel**
PR-1 of PR Push (first in the push.... review and merge first!)

This adds functionality for a working produce page with all CRUD operations available

### CODE REVIEW

File: view/src/components/produce.js [hunk 1]
```diff
@@ -0,0 +1,713 @@
...
+                      </Box>
+                    </CardContent>
+                    <CardActions>
+                      <Button
+                        size="small"
+                        color="primary"
+                        onClick={() => this.handleViewOpen({ produce })}
+                      >
+                        {" "}
+                        View{" "}
```
Comment (andrewhojel) [inline view/src/components/produce.js:632]: I realized why all these spaces were popping up. They are being added by prettier. As a result, I am assuming they are part of some style norm. Thoughts?

File: view/src/components/produce.js [hunk 2]
```diff
@@ -0,0 +1,713 @@
...
+                flexWrap="wrap"
+                p={0}
+                m={0}
+              >
+                <Box p={3}>
+                  <Typography className={classes.pos} color="textSecondary">
+                    Shipping Temperatures in Reefer (°F):
+                  </Typography>
+                  <Typography variant="body2" component="p">
+                    Maintenance Temperature:{" "}
```
Comment (Caffeinix) [inline view/src/components/produce.js:679]: I have a theory about these.  The way HTML works is that any contiguous run of whitespace characters in the source text gets converted to a single space character when rendering.  A newline is a whitespace character.  So, surprisingly:

```
<div>
  <span>Foo</span>
  </span>Bar</span>
</div>
```

renders as "Foo Bar", but:

```
<div><span>Foo</span></span>Bar</span></div>
```

renders as "FooBar".  I think prettier might be trying to reformat your code while still keeping the rendering the same, but either it's getting confused or the original rendering wasn't what you wanted anyway.  In this case, for example, the newline after "Maintenance Temperature:" ought to insert a space into the output even without this...  assuming that the JSX maps to HTML in a reasonable way without compressing away the whitespace.

File: view/src/components/produce.js [hunk 3]
```diff
@@ -0,0 +1,713 @@
...
+      });
+  }
+
+  /**
+   * Takes a produce object as an input and opens a dialog page to
+   * allow the user to update the attributes of the produce object
+   * @param data A produce object
+   */
+  handleEditClickOpen(data) {
+    this.setState({
```
[the remark this answers is not in the document]
Comment (andrewhojel) [reply view/src/components/produce.js:212]: I actually believe it does merge it with the existing state.... For example there are places where I call this.setState() on one state variable and it only changes that variable. Is that what you meant?

File: view/src/components/produce.js (hunk 2 above)
Comment (andrewhojel) [reply view/src/components/produce.js:679]: Intersting!

File: view/src/components/produce.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Caffeinix) [reply view/src/components/produce.js:212]: Yup!  And yeah, if it didn't do that then I think all sorts of things would be broken, so it must.  Thanks for checking!

**Comment by andrewhojel** [comment]
All of the files that aren't produce.js have only been changed by prettier (solely style changes)

File: view/src/components/produce.js [hunk 4]
```diff
@@ -0,0 +1,724 @@
...
+                m={0}
+              >
+                <Box p={3}>
+                  <Typography className={classes.pos} color="textSecondary">
+                    Shipping Temperatures in Reefer (°F):
+                  </Typography>
+                  <Typography variant="body2" component="p">
+                    Maintenance Temperature:
+                    {this.state.shippingMaintenanceTemperatureLow} -
+                    {this.state.shippingMaintenanceTemperatureLow}
```
Comment (Caffeinix) [inline view/src/components/produce.js:692]: Should one of these be shippingMaintenanceTemperatureHigh?

Comment (andrewhojel) [reply view/src/components/produce.js:692]: Good catch! This would have been very confusing

File: view/src/components/produce.js [hunk 5]
```diff
@@ -0,0 +1,724 @@
...
+  },
+  bullet: {
+    display: "inline-block",
+    margin: "0 2px",
+    transform: "scale(0.8)",
+  },
+  pos: {
+    marginBottom: 12,
+  },
+  uiProgess: {
```
Comment (fatimazali) [inline view/src/components/produce.js:74]: maybe this can be uiProgress instead :P

Comment (fatimazali) [reply view/src/components/produce.js:74]: oh wait i just realized this is merged - ill change this in my branch instead. i'm still figuring out the spacing changes btw

**Pull request merged**